src/main/java/seedu/address/model/lesson/Lesson.java

```diff
@@ -25,47 +23,49 @@ public class Lesson {
 
     private final Date date;
     private final Time time;
-    private final List<Student> students;
-    private final Map<Student, Boolean> attendanceMap;
+    private final List<StudentLessonInfo> studentLessonInfoList;
```

Comment:
Nice Idea to Condense All Students, Attendance & Participation into a Single List. It'll make Extending the Code easier in the future.